File: WinForms/CS/DialogsMainDemo/frmMain.cs
namespace DevExpress.XtraDialogs.Demos {
    using DevExpress.XtraEditors;

    public partial class frmMain : DevExpress.DXperience.Demos.RibbonMainForm  {
        protected override void SetFormParam() {
            Icon = Utils.ResourceImageHelper.CreateIconFromResourcesEx("DevExpress.XtraDialogs.Demos.AppIcon.ico", typeof(frmMain).Assembly);
        }
        protected override string ProductName {
            get { return "Xtra Dialogs"; }
        }
        protected override string DemoName {
            get { return "Dialogs Demo (C# code)"; }
        }
        protected override bool SupportAdvancedTitlePainting {
            get { return false; }
        }
        protected override void ShowModule(string name, XtraEditors.GroupControl group, LookAndFeel.DefaultLookAndFeel lookAndFeel) {
            DemosInfo.ShowModule(name, group, RibbonMenuManager);
        }
        protected override void FillNavBar() {
            DXperience.Demos.ModulesInfo.FillAccordionControl(accordionControl1, XtraNavBar.NavBarGroupStyle.SmallIconsList, true, XtraNavBar.NavBarImage.Large);
        }
        protected override bool CanUseEmptyRootPadding(string name) {
            return true;
        }
        protected override void ShowAbout() {
            BaseEdit.About();
        }
    }
}

File: WinForms/VB/DialogsMainDemo/frmMain.vb
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraDialogs.Demos

    Public Partial Class frmMain
        Inherits DXperience.Demos.RibbonMainForm

        Protected Overrides Sub SetFormParam()
            Icon = Utils.ResourceImageHelperCore.CreateIconFromResourcesEx("DevExpress.XtraDialogs.Demos.AppIcon.ico", GetType(frmMain).Assembly)
        End Sub

        Protected Overrides ReadOnly Property ProductName As String
            Get
                Return "Xtra Dialogs"
            End Get
        End Property

        Protected Overrides ReadOnly Property DemoName As String
            Get
                Return "Dialogs Demo (C# code)"
            End Get
        End Property

        Protected Overrides ReadOnly Property SupportAdvancedTitlePainting As Boolean
            Get
                Return False
            End Get
        End Property

        Protected Overrides Sub ShowModule(ByVal name As String, ByVal group As GroupControl, ByVal lookAndFeel As LookAndFeel.DefaultLookAndFeel)
            DemosInfo.ShowModule(name, group, RibbonMenuManager)
        End Sub

        Protected Overrides Sub FillNavBar()
            DXperience.Demos.ModulesInfo.FillAccordionControl(accordionControl1, XtraNavBar.NavBarGroupStyle.SmallIconsList, True, XtraNavBar.NavBarImage.Large)
        End Sub

        Protected Overrides Function CanUseEmptyRootPadding(ByVal name As String) As Boolean
            Return True
        End Function

        Protected Overrides Sub ShowAbout()
            Call BaseEdit.About()
        End Sub
    End Class
End Namespace
